docs/pageserver-tenant-migration.md

```diff
@@ -0,0 +1,20 @@
+## Pageserver tenant migration
+
+### Overview
+
+This feature allows to migrate a timeline from one pageserver to another by utilizing remote storage capability.
+
+### Migration process
+
+Pageserver implements two new http handlers: timeline attach and timeline detach.
+Timeline migration is performed in a following way:
+1. Timeline attach is called on a target pageserver.
+2. For now it is necessary to manually initialize replication stream via callmemaybe call so target pageserver initializes replication from safekeeper (it is desired to avoid this and initialize replication directly in attach handler, but this requires some refactoring (probably [#997](https://github.com/zenithdb/zenith/issues/997)/[#1049](https://github.com/zenithdb/zenith/issues/1049))
+3. Replication state can be tracked via timeline detail pageserver call.
+4. Compute node should be restarted with new pageserver connection string.
+5. Timeline is detached from old pageserver.
+
+
+### Implementation details
+
+Now safekeeper needs to track which pageserver it is replicating to. This introduces complications into replication code and callmemaybe subscription state handling.
```

Comment:
Could you elaborate on the "complications in the replication code"? 

I wonder if we need special treatment of pageserver's feedback in `get_replicas_state()` on safekeeper, given that two streams can exist for the same timeline?

Reply:
By complications I mean additional handling of pageserver connection string and the requirement to track which pageserver is actually a primary one, to avoid reconnections to it (which is not currently implemented but is present in the epic in the follow ups section)

Reply:
> given that two streams can exist for the same timeline?

At a first glance everything looks OK because we keep a Vec of replicas for particular timeline. Do you have some invariants in mind that should be checked?